WordPress/AbstractThemeSniff.php

```diff
@@ -0,0 +1,1352 @@
+	 */
+	public function blacklist_file_check( $themefiles ) {
+		global $sniff_helper;
+		$blacklist = array(
+				'thumbs.db'				=> 'Windows thumbnail store',
+				'desktop.ini'			=> 'windows system file',
+				'project.properties'	=> 'NetBeans Project File',
+				'project.xml'			=> 'NetBeans Project File',
+				'.kpf'					=> 'Komodo Project File',
+				'php.ini'				=> 'PHP server settings file',
+				'dwsync.xml'			=> 'Dreamweaver project file',
+				'error_log'				=> 'PHP error log',
+				'web.config'			=> 'Server settings file',
+				'.sql'					=> 'SQL dump file',
+				'__MACOSX'				=> 'OSX system file',
+				'.lubith'				=> 'Lubith theme generator file',
+				'.zip'					=> 'compressed file',
+
+			);
+		foreach ( $themefiles as $themefile ) {
+			foreach ( $blacklist as $key => $reason ) {
+				if ( false !== strpos( $themefile, $key ) ) {
+					$sniff_helper['files_not_allowed'][] = $key . ' - ' . $reason;
+				}
+			}
+			if ( false !== strpos( $themefile, '\.' ) ) {
+				if ( '\.' !== substr( $themefile, -2 ) && '\..' !== substr( $themefile, -3 ) ) {
+					$sniff_helper['files_not_allowed'][] = 'hidden file or folder:';
+					$sniff_helper['files_not_allowed'][] = $themefile;
+				}
+			}
+		}
+	}
+
+	/**
+	 * Check if screenshot file exists.
+	 * Check screenhot max width 1200px.
+	 * Check screenshot max height 900px.
+	 * Check screenshot ratio 4:3.
+	 *
+	 * @param string $themedir contains the theme directory uri.
+	 */
+	public function get_screenshot_checks( $themedir ) {
+		global $sniff_helper;
+		if ( file_exists( $themedir . '\screenshot.jpg' ) ) {
+			$sniff_helper['screenshot']['found'] = true;
+			$imagefilename = $themedir . '\screenshot.jpg';
+		}
+		if ( file_exists( $themedir . '\screenshot.png' ) ) {
+			$sniff_helper['screenshot']['found'] = true;
+			$imagefilename = $themedir . '\screenshot.png';
+		}
+		if ( false !== $sniff_helper['screenshot']['found'] ) {
+			$image = getimagesize( $imagefilename );
+			if ( is_array( $image ) ) {
+				if ( $image[0] <= 1200 ) {
+					$sniff_helper['screenshot']['less_than_1200_wide'] = true;
+				}
+				if ( $image[1] <= 900 ) {
+					$sniff_helper['screenshot']['less_than_900_high'] = true;
+				}
+
+				if ( 0.76 > $image[1] / $image[0] &&  0.74 < $image[1] / $image[0] ) {
+					$sniff_helper['screenshot']['aspect_ratio_4_by_3'] = true;
+				}
+			} else {
+				$sniff_helper['screenshot']['less_than_1200_wide'] = true;
+				$sniff_helper['screenshot']['less_than_900_high'] = true;
+				$sniff_helper['screenshot']['aspect_ratio_4_by_3'] = true;
+				$sniff_helper['screenshot']['details_not_found'] = true;
+			}
+		}
+	}
+
+	/**
+	 * This function will go through the files and identify all the functions
+	 * that contain a textdomain. Once that is done the textdomains will be extracted
+	 * and then put in an array for later use.
+	 *
+	 * @param string $file_content contains content of file in a single string.
+	 */
+	public function get_textdomains( $file_content ) {
```

Comment:
if only one text domain was allowed (the themeslug) things would be much simpler, and more definitive on the sniff.